### DISCUSSION TITLE
Ensure asset is used only once.

### DISCUSSION THREAD

**Discussion by mGonzim** [nftchef/art-engine#30] [Help + Q&A]
I'm hoping to get some help. Is there a way to choose a folder within the config however to ensure that the assets used within the folder are only used once. I have some programing experience in JS if you can point me in the right direction I would be very appreciative.

**Comment by nftchef** [comment]
This has been suggested before but there is no way currently to do this.

my approach to adding this feature (and/or the commonly requested "only use an item once") would be something like:

- when constructing the available layers probably in `getElements`, add a count property, something like `useCount`
- Then, when selecting the layers in `pickRandomElement` (which is already a bloated function). do a  check on the count flag. 

in your implementation, you would want do set the array of layers to choose from based on a filter of the `useCount` property,
essentially extending the `compatibleLayers` feature to also check for use count
Taking this,
```js
  const compatibleLayers = layer.elements.filter(
    (layer) => !incompatibleDNA.includes(layer.name)
  );
``` 
and adding something like,
```js

  const unusedLayers = layer.elements.filter(layer => layer.useCount === 0)
  const compatibleLayers = layer.elements.filter(
    (layer) => {
      // if there are layers that haven't been used, filter out the used ones
      if (unusedLayers.length > 1) {
        return !incompatibleDNA.includes(layer.name) && layer.useCount == 0;
      } else {
        // if all layers have been used once, only worry about incompatible
        return !incompatibleDNA.includes(layer.name)
      }
    }
  );
```

(untested, non-optimal codefor example)

**Comment by mGonzim** [comment]
You are clearly good at what you do. I have an idea how to implement this
but I would gladly pay for this feature. Any chance this is possible? 🤔
